### ISSUE TITLE
Stackoverflow when accesing FcmSilentData.createdLifeCycle

### ISSUE THREAD

**Issue by Clon1998** [rafaelsetragni/awesome_notifications_fcm#1]
Trying to access `FcmSilentData.createdLifeCycle` field in the `onFcmSilentDataHandle` handler causes an stackoverflow.

Code to reproduce:

```dart
    await _notifyFCM.initialize(
        onFcmTokenHandle: _awesomeNotificationFCMTokenHandler,
        onFcmSilentDataHandle: _awesomeNotificationFCMBackgroundHandler,
        licenseKey: null);
...
  @pragma("vm:entry-point")
  static Future<void> _awesomeNotificationFCMBackgroundHandler(
      FcmSilentData message) async {

    DartPluginRegistrant.ensureInitialized();
    if (message.createdLifeCycle != NotificationLifeCycle.Foreground) {
      print("bg");
    } else {
      print("FOREGROUND");
    }
  }

```

Console Output:
```
W/leraker.androi( 9467): Accessing hidden method Landroid/os/WorkSource;->add(I)Z (unsupported,test-api, reflection, allowed)
W/leraker.androi( 9467): Accessing hidden method Landroid/os/WorkSource;->add(ILjava/lang/String;)Z (unsupported,test-api, reflection, allowed)
W/leraker.androi( 9467): Accessing hidden method Landroid/os/WorkSource;->get(I)I (unsupported, reflection, allowed)
W/leraker.androi( 9467): Accessing hidden method Landroid/os/WorkSource;->getName(I)Ljava/lang/String; (unsupported, reflection, allowed)
D/Android: [Awesome Notifications]( 9467): A new Awesome FCM service has started (AwesomeFcmService:46)
D/Android: [Awesome Notifications]( 9467): received a new fcm push (id: 0:1665002739800278%73e8ddfbf9fd7ecd) (AwesomeFcmService:134)
D/Android: [Awesome Notifications]( 9467): A new silent background service has started (FcmSilentService:23)
I/Android: [Awesome Notifications]( 9467): Initializing Flutter global instance. (FcmDartBackgroundExec:144)
I/Android: [Awesome Notifications]( 9467): Creating background FlutterEngine instance. (FcmDartBackgroundExec:157)
D/EGL_emulation( 9467): eglCreateContext: 0xb400007d27de2290: maj 3 min 0 rcv 3
D/EGL_emulation( 9467): eglCreateContext: 0xb400007d27dd8d50: maj 3 min 0 rcv 3
D/HostConnection( 9467): createUnique: call
D/HostConnection( 9467): HostConnection::get() New Host Connection established 0xb400007d27de2e90, tid 9594
D/HostConnection( 9467): HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_0 
D/EGL_emulation( 9467): eglMakeCurrent: 0xb400007d27dd8d50: ver 3 0 (tinfo 0x7f3a87d280) (first time)
W/MediaSessionCompat( 9467): Couldn't find a unique registered media button receiver in the given context.
D/Android: [Awesome Notifications]( 9467): Awesome Notifications plugin attached to Android 32 (AwesomeNotificationsPlugin:155)
D/Android: [Awesome Notifications]( 9467): Awesome Notifications attached to engine for Android 32 (AwesomeNotificationsPlugin:137)
D/Android: [Awesome Notifications]( 9467):  subscribed to receive FCM events (AwesomeFcmEventsReceiver:51)
D/Android: [Awesome Notifications]( 9467):  subscribed to receive FCM events (AwesomeFcmEventsReceiver:71)
I/Android: [Awesome Notifications]( 9467): Executing background FlutterEngine instance for silent FCM. (FcmDartBackgroundExec:180)
D/EGL_emulation( 9467): app_time_stats: avg=28.29ms min=13.10ms max=63.60ms count=36
I/flutter ( 9467): Awesome Notifications FCM: An error occurred in your background messaging handler:
I/flutter ( 9467): Stack Overflow
I/Android: [Awesome Notifications]( 9467): All silent data fetched. (FcmDartBackgroundExec:249)
I/Android: [Awesome Notifications]( 9467): Shutting down background FlutterEngine instance. (FcmDartBackgroundExec:208)
D/Android: [Awesome Notifications]( 9467): Awesome Notifications plugin detached from Android 32 (AwesomeNotificationsPlugin:180)
D/Android: [Awesome Notifications]( 9467):  unsubscribed from notification events (AwesomeFcmEventsReceiver:59)
D/Android: [Awesome Notifications]( 9467):  unsubscribed from notification events (AwesomeFcmEventsReceiver:79)
I/Android: [Awesome Notifications]( 9467): FlutterEngine instance terminated. (FcmDartBackgroundExec:215)
```

**Comment by rafaelsetragni** [comment]
Yes! I've never seen this happen before. Both enumerators are populated correctly, but when compared they trigger a stack overflow error without any other details. Very strange!

![image](https://user-images.githubusercontent.com/40064496/194165225-a202a918-a6be-4c65-bdd7-8d12c7ba7564.png)

**Comment by rafaelsetragni** [comment]
I got it! It's a bug. Here is the issue:
![image](https://user-images.githubusercontent.com/40064496/194167570-452c55d3-d91f-43fa-b484-d6c3f2589fae.png)
`createdLifeCycle` was pointing to itself. But changing it to:
![image](https://user-images.githubusercontent.com/40064496/194167691-2874c335-7390-4166-b405-a987eb598414.png)
... solves the issue.

I will update the version with this fix in a couple of minutes. Thanks for your help!

**Comment by rafaelsetragni** [comment]
Your fix was published in version of `awesome_notifications: ^0.7.1` and `awesome_notifications_fcm: ^0.7.1`.
Update your dependencies and check it out!

**Comment by Clon1998** [comment]
Can confirm that it works now.
